### DISCUSSION TITLE
Can we still use self-hosted preview gatsby server for wordpress preview in latest versions of gatsby-source-wordpress?

### DISCUSSION THREAD

**Discussion by sanky72** [gatsbyjs/gatsby#36687] [Help] [answered]
Hi everyone, 
I am unable to run preview when I migrated from gatsby-source-wordpress 5 to the latest version. This latest version preview is not working with either wpgatsby older version or latest. 

Wanted to just get clarification if we are using gatsby-source-wordpress version 6^ will be still be able to use a self-hosted instance for wordpress previews?

**Comment by sanky72** [comment]
@TylerBarnes  can you please help regarding this?

**Comment by TylerBarnes** [comment, marked as answer]
@sanky72 self-hosted preview no longer works on the latest WPGatsby, we've unified all our source-plugins to use the same preview service, "Content Sync". We don't have the bandwidth to support multiple kinds of preview in source-wordpress, so the old way was removed. You can read more about Content Sync here https://www.gatsbyjs.com/docs/conceptual/content-sync/. Gatsby Cloud Preview is available on the free tier, so you can try out content sync for free for your previews even if you're not using Gatsby Cloud to build your production site.

**Reply by TylerBarnes** [reply]
There are instructions on setting it up here https://support.gatsbyjs.com/hc/en-us/articles/4410372011027-Installing-Content-Sync-for-WordPress

Let me know if you have any more questions

**Reply by sanky72** [reply]
The problem I'm facing is, due to some infrastructural limitations, I can't use Gatsby Cloud for previews. Is there a way through which I can implement the old way or could you at least direct me through so I don't need to use the Gatsby cloud but can serve preview through a Gatsby dev server?

**Reply by sanky72** [reply]
Or if we can use older version of WPGatsby (which supported self-hosted gatsby server previews) with new gatsby-source-wordpress version ( 6 and above)

**Reply by TylerBarnes** [reply]
The only way to use the old preview is to use an old version of WPGatsby but the problem there is you'll be missing out on bug fixes we've made since then and any we'll make in the future.
Gatsby Cloud is currently the only Content Sync UI implementation but technically it could be used by any hosting platform or by an individual since the underlying API `createNodeManifest` is OSS. It's not an easy thing to implement or to make work reliably though.

**Reply by sanky72** [reply]
Thanks, for the response. I tried using the old version of WPGatsby with new version of gatsby-source-wordpress (6.24.0), but it didn't work.

**Reply by TylerBarnes** [reply]
Yeah there was likely a breaking change so you'd have to use an old source-wordpress too. I wouldn't recommend going that route since it'll make it very hard to upgrade in the future when/if you need to.

**Reply by sanky72** [reply]
Noted. I will try to find a solution for this, thanks for all the help.

**Reply by TylerBarnes** [reply]
Anytime! Let me know if you have any more questions
